src/node/rpc/node_frontend.h

```diff
@@ -332,22 +332,184 @@ namespace ccf
         .install();
 
       auto network_status = [this](auto& args, nlohmann::json&&) {
+        GetNetworkInfo::Out out;
         auto service_view = args.tx.get_read_only_view(network.service);
         auto service_state = service_view->get(0);
         if (service_state.has_value())
         {
-          return make_success(service_state.value().status);
+          out.service_status = service_state.value().status;
+          if (consensus != nullptr)
+          {
+            out.current_view = consensus->get_view();
+
+            auto primary_id = consensus->primary();
+            auto nodes_view = args.tx.get_read_only_view(this->network.nodes);
+            auto info = nodes_view->get(primary_id);
+            if (info)
+            {
+              out.primary_id = primary_id;
+            }
+          }
+          return make_success(out);
         }
         return make_error(
           HTTP_STATUS_NOT_FOUND,
           ccf::errors::ResourceNotFound,
-          "Network status is unknown.");
+          "Service state not available.");
       };
       make_read_only_endpoint(
         "network",
         HTTP_GET,
         json_read_only_adapter(network_status),
         no_auth_required)
+        .set_auto_schema<void, GetNetworkInfo::Out>()
+        .install();
+
+      auto get_nodes = [this](auto& args, nlohmann::json&& params) {
+        const auto in = params.get<GetNodes::In>();
+        GetNodes::Out out;
+
+        auto nodes_view =
```

Comment:
You can probably remove the awkward template here by calling `get_read_only_view` directly on the map instance (i.e. args.tx.get_read_only_view(this->network.nodes)`).

Reply:
Done.